src/Relations/BelongsToMany.php

```diff
@@ -190,10 +190,10 @@ public function attach($id, array $attributes = [], $touch = true)
 
             $query = $this->newRelatedQuery();
 
-            $query->whereIn($this->related->getKeyName(), (array) $id);
+            $query->whereIn($this->relatedKey, (array) $id);
 
             // Attach the new parent id to the related model.
-            $query->push($this->foreignPivotKey, $this->parent->getKey(), true);
+            $query->push($this->foreignPivotKey, $this->parent->{$this->parentKey}, true);
```

Comment:
The `$this->parentKey' is replaced with `$this->parent->getKey()` to respect the custom `parentKey` field.